pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -115,11 +119,7 @@ func setCABundle(caBundle []byte) error {
 	if ok := rootCAs.AppendCertsFromPEM(caBundle); !ok {
 		return errors.New("no certs were appended from caBundle")
 	}
-	client.Transport = &http.Transport{
-		TLSClientConfig: &tls.Config{
-			RootCAs: rootCAs,
-		},
-	}
+	tlsConfig.RootCAs = rootCAs
```

Comment:
🤔 

Does it still work if we do `client.Transport.TLSCClientConfig.RootCAs = rootCAs` instead? (then we could inline `tlsConfig` into `client` in the `var` declaration above?

Reply:
I tried this approach as well but instance is not created by default, TLSclientconfig do not present when we assign align like that. By default it is having the round trip instance not the transport instance.

Reply:
What if above becomes:

```go
var client = http.Client{
	Timeout: 15 * time.Second,
	Transport: &http.Transport{
		TLSClientConfig: &tls.Config{},
	},
}
```

Does that resolve the issue?

Reply:
If we  dont have a tlsConfig variable we will not be able to reference it to update the rootCAs at https://github.com/googleforgames/agones/blob/main/pkg/fleetautoscalers/fleetautoscalers.go#L118-L121.

Reply:
Oh I see the issue now, `http.RoundTripper` is an interface, so you can't access the variables within it.

I was playing with it here: https://go.dev/play/p/C6RTLx9VfWl